core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -411,10 +439,14 @@ Builder scanMetrics(ScanMetrics newScanMetrics) {
       return this;
     }
 
-    DeleteFileIndex build() {
+    private Collection<DeleteFile> planFiles() {
+      if (deleteFiles != null) {
+        return deleteFiles;
+      }
+
       // read all of the matching delete manifests in parallel and accumulate the matching files in
       // a queue
-      Queue<ManifestEntry<DeleteFile>> deleteEntries = new ConcurrentLinkedQueue<>();
```

Comment:
This will be a separate PR. The goal is to move the internal logic to using `DeleteFile` objects directly, instead of relying on manifest entires. That way, we can build an index using a set of files.